### PR TITLE
Bumps the langgraph version and fixes resulting bugs

### PR DESCRIPTION
**Pull request by ghost**
## Context

We want to update langgraph and 0.2.15 seems to make state attributes optional if no node ever amends them. For the documents part of the state, this meant lots of missing keys.

## Changes proposed in this pull request

* Bump langgraph
* Change lots of `state["documents"]` to `state.get("documents")`

## Guidance to review

* Check you're happy that I've bumped the right versions
* Check you're happy with where I've chosen to use `.get()` vs just error

## Relevant links

* [langgraph release history](https://github.com/langchain-ai/langgraph/releases)

## Things to check

- [x] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] [I have run integration tests](https://github.com/i-dot-ai/redbox/actions/runs/10686831863)

### CODE REVIEW

File: redbox-core/redbox/graph/nodes/processes.py [hunk 1]
```diff
@@ -167,7 +170,8 @@ def _set_text(state: RedboxState) -> dict[str, Any]:
 
 
 def clear_documents_process(state: RedboxState) -> dict[str, Any]:
-    return {"documents": {group_id: None for group_id in state["documents"].keys()}}
+    if documents := state.get("documents"):
+        return {"documents": {group_id: None for group_id in documents.keys()}}
```
Comment (gecBurton) [inline redbox-core/redbox/graph/nodes/processes.py:174]:
```suggestion
        return {"documents": {group_id: None for group_id in documents}}
```
nitpick

**Review by jamesrichards4 — APPROVED** [review]
Looks good to me but I don't really understand what LangGraph did here to cause an issue. Be good to have a quick chat through when you're about

File: redbox-core/tests/graph/test_patterns.py [hunk 2]
```diff
@@ -206,8 +206,8 @@ def test_build_merge_pattern(test_case: RedboxChatTestCase, mocker: MockerFixtur
         response = merge(state)
     final_state = RedboxState(response)
 
-    response_documents = [doc for doc in flatten_document_state(final_state["documents"]) if doc is not None]
-    noned_documents = sum([1 for doc in final_state["documents"].values() for v in doc.values() if v is None])
+    response_documents = [doc for doc in flatten_document_state(final_state.get("documents")) if doc is not None]
+    noned_documents = sum([1 for doc in final_state.get("documents", {}).values() for v in doc.values() if v is None])
```
Comment (gecBurton) [inline redbox-core/tests/graph/test_patterns.py:210]:
```suggestion
    noned_documents = sum(1 for doc in final_state.get("documents", {}).values() for v in doc.values() if v is None)
```
nitpick

**Review by gecBurton — APPROVED** [review]
pointless nitpicks added to prove i read it. ignore them and merge this helpful fix!

**Pull request merged**